cluster-tests/tests/smb/test_002_smb_accounts.py

```diff
@@ -0,0 +1,282 @@
+import pytest
+import string
+import secrets
+
+from config import CLUSTER_INFO, CLUSTER_IPS, TIMEOUTS
+from exceptions import JobTimeOut
+from helpers import smb_connection
+from utils import make_request, ssh_test, make_ws_request, wait_on_job
+from pytest_dependency import depends
+from samba import ntstatus, NTSTATUSError
+
+
+SHARE_FUSE_PATH = f'CLUSTER:{CLUSTER_INFO["GLUSTER_VOLUME"]}/smb_share_02'
+alphabet = string.ascii_letters + string.digits
+user_password = ''.join(secrets.choice(alphabet) for i in range(16))
+
+
+@pytest.mark.dependency(name="CLUSTER_SMB_SHARE2_CREATED")
+def test_001_create_clustered_smb_share2(request):
+    global SMB_SHARE_ID
+
+    url = f'http://{CLUSTER_IPS[0]}/api/v2.0/filesystem/mkdir/'
+    res = make_request('post', url, data=SHARE_FUSE_PATH)
+    assert res.status_code == 200, res.text
+
+    url = f'http://{CLUSTER_IPS[0]}/api/v2.0/sharing/smb/'
+    payload = {
+        "comment": "CLUSTER_SMB_SHARE_2",
+        "path": '/smb_share_02',
+        "name": "CL_SMB2",
+        "purpose": "NO_PRESET",
+        "shadowcopy": False,
+        "cluster_volname": CLUSTER_INFO["GLUSTER_VOLUME"]
+    }
+
+    res = make_request('post', url, data=payload)
+    assert res.status_code == 200, res.text
+    SMB_SHARE_ID = res.json()['id']
+
+
+@pytest.mark.parametrize('ip', CLUSTER_IPS)
+def test_002_verify_smb_share2_exists(ip, request):
+    depends(request, ['CLUSTER_SMB_SHARE2_CREATED'])
+
+    url = f'http://{ip}/api/v2.0/sharing/smb?id={SMB_SHARE_ID}'
+    res = make_request('get', url)
+    assert res.status_code == 200, res.text
+    share = res.json()[0]
+    assert share['cluster_volname'] == CLUSTER_INFO["GLUSTER_VOLUME"], str(share)
+    assert share['name'] == 'CL_SMB2', str(share)
+    assert share['path_local'] == SHARE_FUSE_PATH, res.text
+
+
+@pytest.mark.dependency(name="SMB_SERVICE2_STARTED")
+@pytest.mark.parametrize('ip', CLUSTER_IPS)
+def test_004_start_smb_service(ip, request):
```

Comment:
Isn't the start of SMB a cluster-wide event?  e.g. make the call to one node, then confirm the state on all nodes.

Reply:
No. It's not cluster-wide with this particular API. If I were enabling via `ctdb.services`, then it would be, but this is quick-and-dirty test.

Reply:
Simpler to just start and stop by separately hitting all nodes.